validator/tests/unit3/test_scheduler/tests.py

```diff
@@ -31,11 +32,119 @@
 
 from test_scheduler.utils import create_batch
 from test_scheduler.utils import create_intkey_transaction
+from test_scheduler.utils import SchedulerTester
 
 
 LOGGER = logging.getLogger(__name__)
 
 
+class TestSchedulersWithYaml(unittest.TestCase):
+
+    def test_single_block_files_individually(self):
+        """Tests scheduler(s) with yaml files that represent a single
+        block.
+
+        Notes:
+            Tests that the serial scheduler has the correct batch validity
+            and state hash, and that 1 state hash is produced.
+
+        """
+
+        files_with_one_block = ['simple_scheduler_test.yaml']
+        for name in files_with_one_block:
+            file_name = self._path_to_yaml_file(name)
+            serial_scheduler = SerialScheduler(
+                squash_handler=self._context_manager.get_squash_handler(),
+                first_state_hash=self._context_manager.get_first_root(),
+                always_persist=False)
+            tester = SchedulerTester(file_name,
+                                     serial_scheduler,
+                                     self._context_manager)
+            defined_batch_results_dict = tester.batch_results
+            batch_results = tester.run_scheduler()
+            self.assert_batch_validity(
+                defined_batch_results_dict,
+                batch_results)
+            self.assert_one_state_hash(batch_results=batch_results)
```

Comment:
Is there any way to add the expected results to the yaml file?  This might make it easier to define tests in an entirely declarative fashion.  Something to think about for a future PR with more tests.